src/actor.rs

```diff
@@ -0,0 +1,196 @@
+use futures::prelude::*;
+use futures::stream::FuturesUnordered;
+use ractor::{
+    concurrency::Duration, Actor, ActorProcessingErr, ActorRef, BytesConvertable, RpcReplyPort,
+};
+use ractor_cluster::RactorClusterMessage;
+use serde::{Deserialize, Serialize};
+use threshold_crypto::{PublicKeySet, SecretKeyShare, Signature, SignatureShare};
+
+const MPC_RECOVERY_GROUP: &str = "mpc-recovery";
+
+type NodeId = u64;
+type Payload = Vec<u8>;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignResponse {
+    node_id: NodeId,
+    sig_share: SignatureShare,
+}
+
+impl BytesConvertable for SignResponse {
+    fn into_bytes(self) -> Vec<u8> {
+        serde_json::to_vec(&self).unwrap()
+    }
+
+    fn from_bytes(bytes: Vec<u8>) -> Self {
+        serde_json::from_slice(&bytes).unwrap()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignatureResponse {
```

Comment:
It's not clear from the name how is it different from the SignResponce. Let's brainstorm on it or add some comments.

Reply:
Yeah the names are far from perfect for sure. `SignatureResponse` is when you have the full signature ready and can respond to `NewRequest`. `SignResponse` is when you want to respond with your signature share.